src/clients/ping.rs

```diff
@@ -0,0 +1,202 @@
+use super::*;
+use crate::net::Connector;
+use protocol_ping::{Compose, Parse, Request, Response};
+use session::{Buf, BufMut, Buffer};
+use std::borrow::{Borrow, BorrowMut};
+
+/// Launch tasks with one conncetion per task as ping protocol is not mux-enabled.
+pub fn launch_tasks(runtime: &mut Runtime, config: Config, work_receiver: Receiver<WorkItem>) {
+    debug!("launching ping protocol tasks");
+
+    // create one task per "connection"
+    // note: these may be channels instead of connections for multiplexed protocols
+    for _ in 0..config.client().unwrap().poolsize() {
+        for endpoint in config.target().endpoints() {
+            runtime.spawn(task(
+                work_receiver.clone(),
+                endpoint.clone(),
+                config.clone(),
+            ));
+        }
+    }
+}
+
+// a task for ping servers (eg: Pelikan Pingserver)
+#[allow(clippy::slow_vector_initialization)]
+async fn task(work_receiver: Receiver<WorkItem>, endpoint: String, config: Config) -> Result<()> {
+    let connector = Connector::new(&config)?;
+
+    let mut stream = None;
+    let parser = protocol_ping::ResponseParser::new();
+    let mut read_buffer = Buffer::new(4096);
+    let mut write_buffer = Buffer::new(4096);
+
+    while RUNNING.load(Ordering::Relaxed) {
```

Comment:
The connection acquisition part is used in all clients, better abstraction?

Reply:
Will see what I can do there.

Reply:
Per discussion, opened #10 to track this in a follow-up PR.